client/wp-admin-scripts/payment-method-promotions/payment-promotion-row.tsx

```diff
@@ -103,24 +115,37 @@ export const PaymentPromotionRow: React.FC< PaymentPromotionRowProps > = ( {
 		);
 	};
 
+	const onDismiss = () => {
+		setIsVisible( false );
+		updatePaymentGateway( gatewayId, {
+			settings: {
+				is_dismissed: 'yes',
+			},
+		} );
+	};
+
+	if ( ! isVisible ) {
+		return null;
+	}
```

Comment:
I think we should actually move this logic to the `Init.php` file maybe in the `should_register_pre_install_wc_pay_promoted_gateway` function?
Mostly because we actually render the payment gateway in the table, and this will show for a little bit until `promotionGateway` is retrieved.
Could we only make use of this condition when a user dismisses the promotion, so skip this check on first render?

Reply:
Doing it in the above mentioned function would make **WooCommerce Payments** show up in the recommended card below (less intrusive), and might not be a bad thing.

Reply:
I don't think it belongs there for the following reasons:

* Feels like we're conflating the logic for the ability to show with the actual shown logic (`can_view` vs `is_visible`)
* Moving into `should_register_pre_install_wc_pay_promoted_gateway` would mean that we have noticeable delays on dismissal, especially on slower hosts
* I feel dismissing this item only to have it appear in the below list on page refresh would be irritating as a store owner, but cc @elizaan36 or @pmcpinto in case this was your intent

Reply:
Once the WCPay prompt is dismissed it shouldn't appear again (because it would be irritating as you mentioned!). It should only appear in one or the other (main table or "Recommended" card), not both to prevent confusion.